### PR TITLE
Dev2

### PR DESCRIPTION
**Pull request by MateAcademy**
my homework

### CODE REVIEW

**Comment by MateAcademy** [comment]
my homework 08.01.2019

File: src/main/java/homework/home08012019/MainLambdaTemplateStudents.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+					return fr1.getPrice() - fr2.getPrice();
+				}
+
+			}
+		};
+		System.out.println("\nSorted for type with comparator:");
+		fruits.stream().sorted(comparator).forEach(System.out::println);
+
+		System.out.println("\nConvert to mapToInt Stream Price:");
+		int[] prices = fruits.stream().mapToInt(Fruit::getPrice).toArray();
```
Comment (DadizhaO) [inline src/main/java/homework/home08012019/MainLambdaTemplateStudents.java:66]: conduct toArray is not necessary, extra action

File: src/main/java/homework/home08012019/MainLambdaTemplateStudents.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+		System.out.println("\nConvert to mapToLong Stream Price:");
+		long[] prices3 = fruits.stream().mapToLong((Fruit::getPrice)).toArray();
+		System.out.println(Arrays.toString(prices3));
+
+		System.out.println("\nShow vitamins for every fruit");
+		fruits.stream().peek(fr -> System.out.println("For " + fr.getFruitType() + " we show "))
+				.flatMap(fr -> fr.getVitamins().stream()).forEach(System.out::println);
+
+		Stream.of(asList("one","two"),asList("three", "four")).flatMap(x -> x.stream()) ;
```
Comment (DadizhaO) [inline src/main/java/homework/home08012019/MainLambdaTemplateStudents.java:81]: I did not understand why this line is needed
`Stream.of(asList("one","two"),asList("three", "four")).flatMap(x -> x.stream()) ;`

File: src/main/java/homework/home08012019/MainLambdaTemplateStudents.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+		System.out.println(fruits.stream().anyMatch(fr -> fr.getDayToLive() > 50));
+
+		System.out.println("\nCannot any fruits live more 61 days?");
+		System.out.println(fruits.stream().noneMatch(fr -> fr.getDayToLive() > 61));
+
+		System.out.println("\nCan all fruits live more 5 days?");
+		System.out.println(fruits.stream().allMatch(fr -> fr.getDayToLive() >= 5));
+
+		System.out.println("\nMinimum time to live is:");
+		System.out.println(fruits.stream().map(Fruit::getDayToLive).min(Comparator.naturalOrder()).get());
```
Comment (DadizhaO) [inline src/main/java/homework/home08012019/MainLambdaTemplateStudents.java:109]: `System.out.println(fruits.stream().min(Comparator.comparing(Fruit::getDayToLive)).get()`
I think this code is better
map is extra action

**Review by DadizhaO — APPROVED** [review]

File: src/main/java/homework/home08012019/MainLambdaTemplateStudents.java [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+		System.out.println("\nReturns a stream without duplicates:");
+		fruits.stream().distinct().forEach(System.out::println);
+
+		System.out.println("\nConverts every element of stream");
+		fruits.stream().map(fruit -> fruit.getFruitType()+ " " + fruit.getPrice()).forEach(System.out::println);
+
+		System.out.println("\nConverts every element of stream and output Name:");
+		fruits.stream().map(Fruit::getFruitType).forEach(System.out::println);
+
+		//внутри фрукта создать метод который выводит тип фрукта  и цену:
```
Comment (SerhiiPasko) [inline src/main/java/homework/home08012019/MainLambdaTemplateStudents.java:38]: DON'T USE CYRILLIC LETTERS

File: src/main/java/homework/home08012019/MainLambdaTemplateStudents.java (hunk 2 above)
Comment (SerhiiPasko) [reply src/main/java/homework/home08012019/MainLambdaTemplateStudents.java:81]: Becouse, first stream is stream of List\<String\>, after flatMap it is a stream of Strings

File: src/main/java/homework/home08012019/MainLambdaTemplateStudents.java [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+		System.out.println("\nShow forEachOrdered parallel:");
+		fruits.parallelStream().forEachOrdered(System.out::println);
+
+		System.out.println("\nShow array of deliveryDates:");
+		LocalDate[] arrayDate = fruits.stream().map(Fruit::getDeliveryDate).toArray(LocalDate[]::new);
+		System.out.println(Arrays.toString(arrayDate));
+
+		System.out.println("\nShow border timeToLive(bigger then) and price(less then):");
+		Fruit complexFruit = fruits.stream().reduce((fr1, fr2) -> {
+			Fruit result = new Fruit(null, -1, null, -1, null);
```
Comment (SerhiiPasko) [inline src/main/java/homework/home08012019/MainLambdaTemplateStudents.java:126]: You copy-paste my code and did some changes =)

**Review by SerhiiPasko — APPROVED** [review]
Good

File: src/main/java/homework/home08012019/MainLambdaTemplateStudents.java (hunk 4 above)
Comment (MateAcademy) [reply src/main/java/homework/home08012019/MainLambdaTemplateStudents.java:38]: ok

File: src/main/java/homework/home08012019/MainLambdaTemplateStudents.java (hunk 5 above)
Comment (MateAcademy) [reply src/main/java/homework/home08012019/MainLambdaTemplateStudents.java:126]: System.out.println(fruits.stream().map(Fruit::getPrice).reduce(0,Integer::sum));
this is my own code with command:  reduce!

**Comment by MateAcademy** [comment]
good

**Pull request merged**